### PR TITLE
Fix logging

### PR DESCRIPTION
**Pull request by DavidKnott**
### - What I did
Fix logging so that bytes variables can be logged.
Added more tests.
In response to #413 @jacqueswww 

### - How I did it
I added parsing for variables so that they can be logged.
I came up with tests that should be invalid and made sure they were / fixed them if they weren't.

### - How to verify it
Look over the tests I added and make sure they're working as expected.

### - Description for the changelog
None
### - Cute Animal Picture
image

### CODE REVIEW

File: tests/parser/features/test_logging.py [hunk 1]
```diff
@@ -156,12 +156,35 @@ def foo():
...
+    logs = s.head_state.receipts[-1].logs[-1]
+    event_id = u.bytes_to_int(u.sha3(bytes('MyLog(bytes4,bytes29,bytes31)', 'utf-8')))
+    # # Event id is always the first topic
+    assert logs.topics[0] == event_id
+    # # Event id is calculated correctly
+    assert c.translator.event_data[event_id]
+    # # Event abi is created correctly
+    assert c.translator.event_data[event_id] == {'types': ['bytes4', 'bytes29', 'bytes31'], 'name': 'MyLog', 'names': ['arg1', 'arg2', 'arg3'], 'indexed': [True, True, False], 'anonymous': False}
+    # Event is decoded correctly
+    assert c.translator.decode_event(logs.topics, logs.data) ==  {'arg1': b'bar\x00', 'arg2': b'bar\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00', 'arg3': b'foo\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00', '_event_type': b'MyLog'}
```
Comment (fubuloubu) [inline tests/parser/features/test_logging.py]: Is there some sort of helper we can give when working with bytes? Like `b'bar\x00'` I really only care that the first 3 is `bar` and the remaining entries are asserted to be empty. Maybe something like `bytes_helper = lambda str, length: s.decode('utf-8') + bytearray(length-len(str))` so I can write:
```python
assert c.translator.decode_event(logs.topics, logs.data) ==  \
        {'arg1': b'bar\x00', 'arg2': bytes_helper('bar', 29), 'arg3': bytes_helper('foo', 31), '_event_type': b'MyLog'}
```

Comment (fubuloubu) [inline tests/parser/features/test_logging.py]: Is there some sort of helper we can give when working with bytes? Like `b'bar\x00'` I really only care that the first 3 is `bar` and the remaining entries are asserted to be empty. Maybe something like `bytes_helper = lambda str, length: s.decode('utf-8') + bytearray(length-len(str))` so I can write:
```python
assert c.translator.decode_event(logs.topics, logs.data) ==  \
        {'arg1': b'bar\x00', 'arg2': bytes_helper('bar', 29), 'arg3': bytes_helper('foo', 31), '_event_type': b'MyLog'}
```

Comment (jacqueswww) [reply tests/parser/features/test_logging.py]: @fubuloubu I think a custom assert is what I have used for similar functionality in the past (this was with standard python testcases). https://pytest.readthedocs.io/en/2.1.0/assert.html#defining-your-own-assertion-comparison (this seems to be the pytest way :smile:  )

Comment (DavidKnott) [reply tests/parser/features/test_logging.py]: Okay, I'll do it that way.

Comment (fubuloubu) [reply tests/parser/features/test_logging.py]: @jacqueswww Not sure I see how a custom assert would help, but a fixture that lets us do this testing easier is what I'm aiming for

Comment (jacqueswww) [reply tests/parser/features/test_logging.py]: @fubuloubu fixture might be an option too yes, using custom assert will allow you to do something like fuzzy match 2 dicts, only comparing the first 3 chars of the values.

**Comment by DavidKnott** [comment]
@jacqueswww I added a failing test because you can't index `>bytes32` and although you can log  `>bytes32` with data neither Solidity nor Viper currently support it.

**Pull request merged**